index.js

```diff
@@ -1,45 +1,90 @@
 #!/usr/bin/env node
 const { execSync } = require('child_process')
+
 const yargs = require('yargs/yargs')
+const chalk = require('chalk')
 
 // from @eqworks/release convention
 const R = /(?<cat>\S+?)(\/(?<t2>\S+))? - (?<title>.*)/
 
+const SEV_COLOR = {
+  log: 'green',
+  warn: 'yellow',
+  error: 'red',
+}
+const SEV_PRE = {
+  log: '✓',
+  warn: '⚠',
+  error: '✖',
+}
+const _log = (sev = 'log') => m => (console[sev])((chalk[SEV_COLOR[sev]])(`${SEV_PRE[sev]} ${m}`))
+const log = _log()
+const warn = _log('warn')
+const error = _log('error')
+const isGood = m => m.match(R)
+const isBad = m => !isGood(m)
+
 if (require.main === module) {
-  const { argv } = yargs(process.argv.slice(2))
-  const verbose = argv.verbose || argv.v
-  const base = argv.base || argv.b || 'origin/main'
+  const { argv } = yargs(process.argv.slice(2)).options({
+    verbose: {
+      alias: 'v',
+      description: 'verbose/debug flag',
+      type: 'boolean',
+      default: false,
+    },
+    base: {
+      alias: 'b',
+      description: 'base git ref to compare with the head ref',
+      type: 'string',
+      default: 'origin/main',
+    },
+    head: {
+      alias: 'h',
+      description: 'head git ref to compare with the base ref',
+      type: 'string',
+      default: 'HEAD',
+    },
+    fetch: {
+      description: 'perform a git fetch before commit-watch',
+      type: 'boolean',
+      default: false,
+    },
+  })
 
   if (argv.fetch) {
-    execSync('git fetch --prune')
+    execSync('git fetch')
   }
 
+  const { verbose, base, head } = argv
+
   try {
-    const messages = execSync(`git log --no-merges --format='%s' ${base}..HEAD`, { stdio: 'pipe' }).toString().trim()
+    const gitLog = `git log --no-merges --format='%s' ${base}..${head}`
+    const messages = execSync(gitLog, { stdio: 'pipe' }).toString().trim()
     if (!messages) {
       if (verbose) {
-        console.warn(`No commits between ${base}..HEAD`)
+        log(`No commits between ${base}..${head}`)
```

Comment:
preference: it's a little confusing to see this message in green with a checkmark since it's more of a neutral statement than commits being evaluated and confirmed to be all good. 

maybe change 'log' in `SEV_COLOR` (should be `COLOUR` cause 🇨🇦 ) and `SEV_PRE` to 'success' and add another key 'log' with a diff colour (or as white) and '-' icon or something.

Reply:
![Screen Shot 2021-05-20 at 2 47 19 PM](https://user-images.githubusercontent.com/42495034/119032674-72e9b680-b97a-11eb-88b0-2981bef6cbb2.png)

Reply:
there's no `console.success()` i believe (accessing through indexing `console['log']` basically)